src/Microsoft.ML.Transforms/MissingValueReplacing.cs

```diff
@@ -595,20 +597,20 @@ public Mapper(MissingValueReplacingTransformer parent, Schema inputSchema)
                 {
                     var type = _infos[i].TypeSrc;
                     if (type is VectorType vectorType)
-                        type = new VectorType((PrimitiveType)type.ItemType, vectorType);
-                    var repType = _parent._repIsDefault[i] != null ? _parent._replaceTypes[i] : _parent._replaceTypes[i].ItemType;
-                    if (!type.ItemType.Equals(repType.ItemType))
+                        type = new VectorType(vectorType.ItemType, vectorType);
+                    var repType = _parent._repIsDefault[i] != null ? _parent._replaceTypes[i] : _parent._replaceTypes[i].GetItemType();
+                    if (!type.GetItemType().Equals(repType.GetItemType()))
                         throw Host.ExceptParam(nameof(InputSchema), "Column '{0}' item type '{1}' does not match expected ColumnType of '{2}'",
-                            _infos[i].Source, _parent._replaceTypes[i].ItemType.ToString(), _infos[i].TypeSrc);
+                            _infos[i].Source, _parent._replaceTypes[i].GetItemType().ToString(), _infos[i].TypeSrc);
                     // If type is a vector and the value is not either a scalar or a vector of the same size, throw an error.
-                    if (repType.IsVector)
+                    if (repType is VectorType repVectorType)
                     {
-                        if (!type.IsVector)
+                        if (!(type is VectorType vectorType2))
```

Comment:
>vectorType2 [](start = 49, length = 11)

you already casted type to VectorType, why you have this vectorType2?

Reply:
I don't think the test on line 599 is available any more, since that `vectorType` would only have scope inside the conditional? And anyway, that `type` is now potentially reassigned to something else, from what I see?